pkg/continuoustest/metrics.go

```diff
@@ -10,45 +10,45 @@ import (
 // TestMetrics holds generic metrics tracked by tests. The common metrics are used to enforce the same
 // metric names and labels to track the same information across different tests.
 type TestMetrics struct {
-	writesTotal                  prometheus.Counter
+	writesTotal                  *prometheus.CounterVec
 	writesFailedTotal            *prometheus.CounterVec
-	queriesTotal                 prometheus.Counter
-	queriesFailedTotal           prometheus.Counter
-	queryResultChecksTotal       prometheus.Counter
-	queryResultChecksFailedTotal prometheus.Counter
+	queriesTotal                 *prometheus.CounterVec
+	queriesFailedTotal           *prometheus.CounterVec
+	queryResultChecksTotal       *prometheus.CounterVec
+	queryResultChecksFailedTotal *prometheus.CounterVec
 }
 
 func NewTestMetrics(testName string, reg prometheus.Registerer) *TestMetrics {
 	return &TestMetrics{
-		writesTotal: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+		writesTotal: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
 			Name:        "mimir_continuous_test_writes_total",
 			Help:        "Total number of attempted write requests.",
 			ConstLabels: map[string]string{"test": testName},
-		}),
+		}, []string{"metric_name"}),
```

Comment:
As discussed , we should call the additional label "type".
```suggestion
		}, []string{"type"}),
```

Also in all cases bellow